### PR TITLE
feat: use RNG when dp routing targets are tied; override no-assume-kv-reuse for decode requests

### PR DESCRIPTION
**Pull request by karen-sy**
#### Overview:

Two features:
(1) When routing targets are tied in tree_size, it makes a selection based on a hashmap id. The existing comment suggests this is pseudorandom, but empirically it seems that these hashmap id rankings are actually quite deterministic -- there is load imbalance among `(worker_id, dp_rank)`s in dp rank routing scenarios.
(2) Add an override for decode requests such that we always do `no-assume-kv-reuse` for more accurate load balancing

#### Details:

Modifies tiebreaking logic in router scheduler ; adds a assume-kv-reuse field to override config

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes GitHub issue: #xxx

## Summary by CodeRabbit

* **New Features**
  * Added support for per-request routing configuration overrides to enable fine-grained control over caching behavior and routing decisions for individual requests.

* **Improvements**
  * Enhanced worker selection algorithm with improved tie-breaking logic to provide better load distribution and fairness when multiple workers have equivalent performance metrics.

### CODE REVIEW

File: lib/llm/src/kv_router.rs [hunk 1]
```diff
@@ -449,7 +449,7 @@ impl KvRouter {
 
         let maybe_seq_hashes = self
             .kv_router_config
-            .compute_seq_hashes_for_tracking(tokens, self.block_size);
+            .compute_seq_hashes_for_tracking(tokens, self.block_size, None);
```
Comment (PeaBrane) [inline lib/llm/src/kv_router.rs]: need the config override here and below i think also

File: lib/llm/src/kv_router/scheduler.rs [hunk 2]
```diff
@@ -540,20 +540,31 @@ impl WorkerSelector for DefaultWorkerSelector {
         let candidates = softmax_sample(&worker_logits, temperature);
 
         // If multiple candidates (tied), use tree size as tie-breaker
-        // If tree sizes are also equal, min_by_key uses HashMap iteration order (pseudo-random)
+        // If tree sizes are also equal, use random selection to avoid bias
         let best_worker = if candidates.len() > 1 {
             tracing::info!("Multiple workers tied with same logit, using tree size as tie-breaker");
-            *candidates
+            let tree_sizes: Vec<_> = candidates
```
Comment (PeaBrane) [inline lib/llm/src/kv_router/scheduler.rs]: can reuse `tree_sizes` something like 

```
let best_worker = if candidates.len() > 1 {
    let tree_sizes: Vec<(usize, &Worker)> = candidates
        .iter()
        .map(|w| (request.overlaps.tree_sizes.get(w).copied().unwrap_or(0), w))
        .collect();

    if tree_sizes.iter().all(|(s, _)| *s == tree_sizes[0].0) {
        let idx = rand::rng().random_range(0..candidates.len());
        candidates[idx]
    } else {
        *tree_sizes.iter().min_by_key(|(s, _)| *s).unwrap().1
    }
}
```

File: lib/llm/src/kv_router.rs (hunk 1 above)
Comment (karen-sy) [reply lib/llm/src/kv_router.rs]: so true

File: lib/bindings/c/src/lib.rs
Comment (karen-sy) [inline lib/bindings/c/src/lib.rs:1]: @PeaBrane Can I leave this as `None`? AI thinks there's no codepath/use case for this binding anyways, but could be wrong. (Also same question for `.get_potential_loads(&token_ids, None)` in `kv.rs` below)

Comment (PeaBrane) [reply lib/bindings/c/src/lib.rs:1]: sure

**Review by PeaBrane — APPROVED** [review]
